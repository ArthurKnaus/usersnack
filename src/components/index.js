import Layout from './Layout';
import PizzaListItem from './PizzaListItem';
import PizzaDescription from './PizzaDescription';
import ChipSelect from './ChipSelect';
import PriceTable from './PriceTable';
import OrderForm from './OrderForm';

export {
  Layout,
  PizzaListItem,
  PizzaDescription,
  ChipSelect,
  PriceTable,
  OrderForm,
};
