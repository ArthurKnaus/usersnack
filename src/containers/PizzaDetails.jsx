import { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { RouteComponentProps } from 'react-router';

import { Card, CardHeader, CardContent, CardMedia, Typography, IconButton } from 'material-ui';
import Divider from 'material-ui/Divider';
import { withStyles } from 'material-ui/styles';
import NavigateBeforeIcon from 'material-ui-icons/NavigateBefore';

import { getPizzaById } from '../requests/pizzas';
import { setAppTitle } from '../store/app/actions';

const styles = () => ({
  card: {
    position: 'relative',
  },
  cardHeader: {
    paddingTop: 0,
    paddingLeft: 24,
  },
  backButton: {
    position: 'absolute',
    height: 32,
    width: 32,
    top: 16,
    left: 8,
  },
  descriptionWrapper: {
    display: 'flex',
  },
  description: {
    flex: '1 1 auto',
  },
  image: {
    minWidth: 167,
    minHeight: 167,
    maxHeight: 167,
  },
});

class PizzaDetails extends Component {
  constructor() {
    super();
    this.state = {
      pizza: null,
    };
  }

  componentWillMount() {
    this.setState({
      pizza: getPizzaById(this.props.match.params.id),
    }, () => {
      const newTitle = `Usersnack - ${this.state.pizza.name}`;
      this.props.setAppTitle(newTitle);
    });
  }

  render() {
    const { classes } = this.props;
    const { pizza } = this.state;

    if (!pizza) return null;

    return (
      <Card className={classes.card}>
        <CardContent className={classes.descriptionWrapper}>
          <div className={classes.description}>
            <IconButton
              className={classes.backButton}
              color="primary"
              onClick={() => this.props.history.push('/pizza')}
            >
              <NavigateBeforeIcon />
            </IconButton>
            <CardHeader
              className={classes.cardHeader}
              title={pizza.name}
              subheader={`${pizza.price} â‚¬`}
            />
            <Typography variant="subheading">
              Ingredients:
            </Typography>
            <Typography variant="body2">
              {pizza.ingredients.join(', ')}
            </Typography>
          </div>
          <CardMedia
            className={classes.image}
            image={`/static/${pizza.img}`}
            title={pizza.name}
          />
        </CardContent>
        <Divider light />
        <Typography variant="subheading">
          TODO: Form
        </Typography>
      </Card>
    );
  }
}

PizzaDetails.propTypes = {
  setAppTitle: PropTypes.func.isRequired,
  ...RouteComponentProps,
};

const mapDispatchToProps = dispatch => ({
  setAppTitle: title => dispatch(setAppTitle(title)),
});

export default connect(null, mapDispatchToProps)(withStyles(styles)(PizzaDetails));
