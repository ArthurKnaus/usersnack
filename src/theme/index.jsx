import React from 'react';

import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles';

const themeObject = {
  palette: {
    primary: {
      light: '#5e92f3',
      main: '#1565c0',
      dark: '#003c8f',
      contrastText: '#fff',
    },
    secondary: {
      light: '#ffd149',
      main: '#ffa000',
      dark: '#c67100',
      contrastText: '#000',
    },
  },
};

const ThemeProvider = props => (
  <MuiThemeProvider
    {...props}
    theme={createMuiTheme(themeObject)}
  />
);

export default ThemeProvider;
