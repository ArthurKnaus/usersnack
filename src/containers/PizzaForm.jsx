import { Component } from 'react';
import { PropTypes } from 'prop-types';

import { CardContent, Typography } from 'material-ui';

import { ChipSelect, PriceTable, OrderForm } from '../components';

import { getAllExtras } from '../requests/extras';


class PizzaForm extends Component {
  constructor(props) {
    super(props);

    this.handleItemChange = this.handleItemChange.bind(this);
    this.handleFormValueChange = this.handleFormValueChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.getAlertText = this.getAlertText.bind(this);

    this.state = {
      extras: [],
      items: [],
      selectedItems: [],
      formValues: {
        name: '',
        address: '',
      },
      formErrors: {
        name: false,
        address: false,
      },
    };
  }

  componentWillMount() {
    const extras = getAllExtras();
    const items = extras.map(extra => extra.name);
    this.setState({
      extras,
      items,
    });
  }

  getAlertText() {
    const sb = [];
    sb.push(`You have successfully ordered a ${this.props.pizza.name}`);
    if (this.state.selectedItems.length > 0) {
      sb.push('Extras:');
      this.state.selectedItems.forEach(e => sb.push(` -${e}`));
    }
    return sb.join('\n');
  }

  handleItemChange(selectedItems) {
    this.setState({
      selectedItems,
    });
  }

  handleFormValueChange(key, value) {
    const { formValues, formErrors } = this.state;
    formValues[key] = value;
    formErrors[key] = value === '';
    this.setState({
      formValues,
      formErrors,
    });
  }

  handleSubmit() {
    const { formValues, formErrors } = this.state;
    let hasError = false;
    Object.keys(formValues).forEach((key) => {
      formErrors[key] = formValues[key] === '';
      hasError = hasError || formErrors[key];
    });
    if (hasError) {
      this.setState({
        formErrors,
      });
    } else {
      alert(this.getAlertText()); //eslint-disable-line
    }
  }

  render() {
    const { pizza } = this.props;

    return (
      <CardContent>
        <Typography variant="subheading" color="textSecondary">
          Choose extras for {pizza.name}:
        </Typography>
        <ChipSelect
          id="pizza-extras"
          label="Extras"
          items={this.state.items}
          value={this.state.selectedItems}
          onChange={this.handleItemChange}
        />
        <PriceTable
          extras={this.state.extras.filter(e =>
            this.state.selectedItems.indexOf(e.name) !== -1)
          }
          basePrice={pizza.price}
        />
        <OrderForm
          values={this.state.formValues}
          errors={this.state.formErrors}
          onValueChange={this.handleFormValueChange}
          onSubmit={this.handleSubmit}
        />
      </CardContent>
    );
  }
}

PizzaForm.propTypes = {
  pizza: PropTypes.object.isRequired,
};

export default PizzaForm;
