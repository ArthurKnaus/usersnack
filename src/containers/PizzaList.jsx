import { Component } from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import PropTypes from 'prop-types';

import { PizzaListItem } from '../components';
import { getAllPizzas } from '../requests/pizzas';
import { setAppTitle } from '../store/app/actions';

class PizzaList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pizzas: [],
    };
  }

  componentWillMount() {
    this.setState({
      pizzas: getAllPizzas(),
    });
    this.props.setAppTitle('Usersnack - Pizza Service');
  }

  render() {
    const { pizzas } = this.state;

    if (pizzas.length < 1) return null;
    return (
      <div>
        { pizzas.map(pizza => (
          <PizzaListItem
            key={pizza.id}
            pizza={pizza}
            onClick={() => this.props.history.push(`/pizzas/${pizza.id}`)}
          />
        )) }
      </div>
    );
  }
}

PizzaList.propTypes = {
  history: PropTypes.object.isRequired, //eslint-disable-line
  setAppTitle: PropTypes.func.isRequired,
  ...RouteComponentProps,
};

const mapDispatchToProps = dispatch => ({
  setAppTitle: title => dispatch(setAppTitle(title)),
});

export default connect(null, mapDispatchToProps)(PizzaList);
