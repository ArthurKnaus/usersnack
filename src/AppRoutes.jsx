import { BrowserRouter } from 'react-router-dom';
import { Switch, Route, Redirect } from 'react-router';

import { PizzaDetails, PizzaList } from './containers';

const AppRoutes = () => (
  <BrowserRouter>
    <Switch>
      { /* ---- PIZZAS ---- */ }
      <Route path="/pizzas" exact component={PizzaList} />
      <Route path="/pizzas/:id" exact component={PizzaDetails} />

      { /* ---- DEFAULT REDIRECT ---- */ }
      <Redirect from="/" to="/pizzas" />
    </Switch>
  </BrowserRouter>
);

export default AppRoutes;
