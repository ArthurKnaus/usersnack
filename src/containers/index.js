import PizzaDetails from './PizzaDetails';
import PizzaList from './PizzaList';

export {
  PizzaDetails,
  PizzaList,
};
