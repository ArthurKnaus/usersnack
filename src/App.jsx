import { Provider as StoreProvider } from 'react-redux';
import ThemeProvider from './theme';
import store from './store';

import { Layout } from './components';
import AppRoutes from './AppRoutes';

const App = () => (
  <ThemeProvider>
    <StoreProvider store={store}>
      <Layout>
        <AppRoutes />
      </Layout>
    </StoreProvider>
  </ThemeProvider>
);

export default App;
