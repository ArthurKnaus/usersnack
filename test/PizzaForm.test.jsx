import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { PizzaForm } from '../src/containers';

Enzyme.configure({ adapter: new Adapter() });

const pizza = {
  id: 2,
  name: 'Mighty Meaty',
  price: 16.90,
  ingredients: ['tomato', 'pepperoni', 'ham', 'onion', 'mushrooms', 'sausage'],
  img: 'mighty-meaty-pizza.jpg',
};

describe('PizzaForm', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<PizzaForm pizza={pizza} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with pizza', () => {
    const wrapper = shallow(<PizzaForm pizza={pizza} />);
    expect(wrapper.exists()).toBe(true);
  });
});
