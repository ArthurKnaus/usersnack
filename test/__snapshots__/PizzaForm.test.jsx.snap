// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaForm should match snapshot 1`] = `
<div
  className="MuiCardContent-root-1"
>
  <h3
    className="MuiTypography-root-2 MuiTypography-subheading-9 MuiTypography-colorTextSecondary-24"
  >
    Choose extras for 
    Mighty Meaty
    :
  </h3>
  <div
    className="MuiFormControl-root-30 ChipSelect-formControl-26"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-40 MuiInputLabel-root-34 MuiInputLabel-formControl-35 MuiInputLabel-animated-38"
      data-shrink={false}
      htmlFor="pizza-extras"
    >
      Extras
    </label>
    <div
      className="MuiInput-root-49 MuiInput-formControl-50 MuiInput-inkbar-51 MuiInput-underline-55"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiSelect-root-44"
      >
        <div
          aria-haspopup="true"
          aria-owns={null}
          aria-pressed="false"
          className="MuiSelect-select-45 MuiSelect-selectMenu-46 MuiInput-input-58"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={undefined}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        />
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          defaultValue={undefined}
          id="pizza-extras"
          name={undefined}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-64 MuiSelect-icon-48"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <table
    className="MuiTable-root-75"
  >
    <thead>
      <tr
        className="MuiTableRow-root-76 MuiTableRow-typeHead-77"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-81 MuiTableCell-paddingDefault-86 MuiTableCell-typeHead-83"
        >
          Item
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-81 MuiTableCell-numeric-82 MuiTableCell-paddingDefault-86 MuiTableCell-typeHead-83"
        >
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MuiTableRow-root-76"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-81 MuiTableCell-paddingDefault-86 MuiTableCell-typeBody-84"
        >
          Base price
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-81 MuiTableCell-numeric-82 MuiTableCell-paddingDefault-86 MuiTableCell-typeBody-84"
        >
          €16.90
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr
        className="MuiTableRow-root-76 MuiTableRow-typeFooter-78"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-81 MuiTableCell-paddingDefault-86 MuiTableCell-typeFooter-85"
        >
          Total
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-81 MuiTableCell-numeric-82 MuiTableCell-paddingDefault-86 MuiTableCell-typeFooter-85"
        >
          €16.90
        </td>
      </tr>
    </tfoot>
  </table>
  <div
    className="OrderForm-container-89"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-30 MuiFormControl-marginNormal-31 OrderForm-input-90"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-40 MuiInputLabel-root-34 MuiInputLabel-formControl-35 MuiInputLabel-animated-38"
        data-shrink={false}
        htmlFor="name"
      >
        Name
        <span
          className=""
        >
           *
        </span>
      </label>
      <div
        className="MuiInput-root-49 MuiInput-formControl-50 MuiInput-inkbar-51 MuiInput-underline-55"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={true}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-58"
          defaultValue={undefined}
          disabled={false}
          id="name"
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={true}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-30 MuiFormControl-marginNormal-31 OrderForm-input-90"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-40 MuiInputLabel-root-34 MuiInputLabel-formControl-35 MuiInputLabel-animated-38"
        data-shrink={false}
        htmlFor="address"
      >
        Address
        <span
          className=""
        >
           *
        </span>
      </label>
      <div
        className="MuiInput-root-49 MuiInput-formControl-50 MuiInput-inkbar-51 MuiInput-underline-55"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={true}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-58"
          defaultValue={undefined}
          disabled={false}
          id="address"
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={true}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root-108 MuiButton-root-93 MuiButton-raised-98 MuiButton-raisedPrimary-100 OrderForm-button-91"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-94"
      >
        Order
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-64 OrderForm-icon-92"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-110"
      />
    </button>
  </div>
</div>
`;
